deepinv/models/flowunet.py

```diff
@@ -0,0 +1,547 @@
+                in_ch,
+                temb_ch=temb_ch,
+                out_ch=in_ch,
+                dropout=dropout,
+                act=act,
+                normalize=normalize,
+            )
+        ]
+        self.mid_modules = nn.ModuleList(mid_modules)
+
+        # Upsampling
+        up_modules = []
+        for i_level in reversed(range(num_resolutions)):
+            # Residual blocks for this resolution
+            block_modules = {}
+            out_ch = ch * ch_mult[i_level]
+            for i_block in range(num_res_blocks + 1):
+                block_modules["{}a_{}a_block".format(i_level, i_block)] = ResidualBlock(
+                    in_ch=in_ch + unet_chs.pop(),
+                    temb_ch=temb_ch,
+                    out_ch=out_ch,
+                    dropout=dropout,
+                    act=act,
+                    normalize=normalize,
+                )
+                if in_ht in attn_resolutions:
+                    block_modules["{}a_{}b_attn".format(i_level, i_block)] = (
+                        SelfAttention(out_ch, normalize=normalize)
+                    )
+                in_ch = out_ch
+            # Upsample
+            if i_level != 0:
+                block_modules["{}b_upsample".format(i_level)] = upsample(
+                    out_ch, with_conv=resamp_with_conv
+                )
+                in_ht *= 2
+            # convert list of modules to a module list, and append to a list
+            up_modules += [nn.ModuleDict(block_modules)]
+        # conver to a module list
+        self.up_modules = nn.ModuleList(up_modules)
+        assert not unet_chs
+
+        # End
+        self.end_conv = nn.Sequential(
+            normalize(in_ch),
+            self.act,
+            conv2d(in_ch, output_channels, init_scale=0.0),
+        )
+
+        print("FlowUNet initialized", pretrained)
+        if pretrained is not None:
+            if pretrained == "download":
+                name = "celeba"
+                # TODO: handle different datasets
+                # TODO: check architecture
+                url = get_weights_from_drive(name)
+                # ckpt = torch.hub.load_state_dict_from_url(
+                #     url, map_location=lambda storage, loc: storage, file_name=name
+                # )
+                # TODO: fix import from drive
+                import gdown
+                print(f"Downloading pretrained model from {url}")
+                gdown.download(url,  './model_final_celeba.pt')
+                ckpt = torch.load('./model_final_celeba.pt',
+                                  map_location=torch.device(self.device))
+            else:
+                ckpt = torch.load(
+                    "./model_final_celeba.pt", map_location=torch.device(self.device)
+                )
+                # ckpt = torch.load(
+                #     pretrained, map_location=lambda storage, loc: storage)
+
+            self.load_state_dict(ckpt, strict=True)
```

Comment:
```suggestion
        if pretrained is not None:
            if pretrained == "download":
                if dataset_name == 'cat':
                    name = "ot_fm_cat.pt"
                elif dataset_name == 'celeba':
                    name = "ot_fm_celeba.pt"
                url = get_weights_url(model_name="drunet", file_name=name)
                ckpt_otfm = torch.hub.load_state_dict_from_url(
                    url, map_location=lambda storage, loc: storage, file_name=name
                )
            else:
                ckpt_otfm = torch.load(
                    pretrained, map_location=lambda storage, loc: storage
                )

            self.load_state_dict(ckpt_otfm, strict=True)
```

Something like this should now work (as done in [here](https://github.com/deepinv/deepinv/blob/main/deepinv/models/drunet.py#L135C1-L150C59))